refresh.lua

```diff
@@ -0,0 +1,19 @@
+-- refresh.lua: => Arguments: [value, ttl]
+-- refresh.lua refreshes provided keys's ttls if all their values match the input. 
+
+-- Check all keys values matches provided input.
+local values = redis.call("mget", unpack(KEYS))
+for i, _ in ipairs(KEYS) do
+	if values[i] ~= ARGV[1] then
+		return redis.error_reply("redislock: not obtained")
+	end
+end
+
+-- Update keys ttls.
+for _, key in ipairs(KEYS) do
+	if redis.call("pexpire", key, ARGV[2]) ~= 1 then
```

Comment:
After reviewing this I actually becoming a little more hesitant to merge this. The problem is that it's actually not trivial to manage multiple locks. Assuming you hold a lock on 3 keys and you are refreshing, the first one succeeds, the second one fails, the third one is never attempted. This behaviour will not be particularly clear to a user. Shouldn't we abandon the first lock in this case too?

Reply:
I am starting to wonder if it's better to allow operating multiple `Lock` instances through a Redis pipeline. This way, each `Lock` holds its own unique state and the user can decide what to do on failures.

Reply:
It seems to me that if something fails in the process that makes that some of the keys that the user wants to lock is not done successfully, everything has to fail. The lock process with many keys must be easy for the user (while maintaining security).

Reply:
Agree, it should be either all or nothing which is not exactly easy, given that you cannot just revert to the previous state mid-script.

Reply:
Do you think that implementing `MultiLock` will make it easier to roll back in case of failure?

Reply:
Can the pexpire fail ? it seems to me that the command cannot fail in the key exists: https://redis.io/commands/pexpire/
Assuming the user must owns all the lock (first loop of this script), then pexpire cannot half-fail.

Reply:
I updated all scripts using pexpire and dont check for pexpire error anymore.
What do you think ?
Of course a server or connexion issue can always be involved, but I'm not sure its possible while executing the lua script on the server.